### PR TITLE
Improve docs for equality and DataFrames

### PR DESCRIPTION
**Pull request by spurll**
Adds documentation detailing some special considerations to be made when comparing `Interval`s with `AnchoredInterval`s.

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -123,6 +123,30 @@ julia> ZonedDateTime(he)
 
 ### Comparisons
 
+#### Equality
+
+Two `AbstractInterval`s are considered equal if they have identical left and right
+endpoints (taking `Inclusivity` into account):
```
Comment (omus) [inline docs/src/index.md]: It would be really great to have docstrings for `LeftEndpoint` and `RightEndpoint`

File: docs/src/index.md [hunk 2]
```diff
@@ -146,17 +170,144 @@ julia> 0..10 ≪ 11..20
...
+│ 7   │ (2016-11-16 HE07] │ 7    │ 'g' │
+│ 8   │ (2016-11-16 HE08] │ 8    │ 'h' │
+│ 9   │ (2016-11-16 HE09] │ 9    │ 'i' │
+│ 10  │ (2016-11-16 HE10] │ 10   │ 'j' │
+│ 11  │ (2016-11-16 HE11] │ 11   │ 'k' │
+│ 12  │ (2016-11-16 HE12] │ 12   │ 'l' │
+
+julia> join(df2, df3; on=:time)
+0×3 DataFrames.DataFrame
+
```
Comment (omus) [inline docs/src/index.md]: Extra whitespace? I guess this is actually what is produced.

Comment (spurll) [reply docs/src/index.md]: Correct! It looked weird when I removed it, so I left it in.

File: docs/src/index.md (hunk 1 above)
Comment (omus) [reply docs/src/index.md]: Ya, that guy is a slacker ;). I'll open an issue for this as I don't think I'll have the time at the moment: https://github.com/invenia/Intervals.jl/issues/28

**Pull request merged**